#! /bin/sh /usr/share/dpatch/dpatch-run
## 85_rebuild_python_ui.dpatch by  <till.kamppeter@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: compiling .ui files to .py

@DPATCH@
diff -urNad hplip-2.8.9~/Makefile.am hplip-2.8.9/Makefile.am
--- hplip-2.8.9~/Makefile.am	2008-09-17 03:18:48.000000000 +1000
+++ hplip-2.8.9/Makefile.am	2008-10-17 13:10:30.000000000 +1100
@@ -4,6 +4,10 @@
 #  (c) 2004-2007 Copyright Hewlett-Packard Development Company, LP
 #
 
+# Generic rules
+%.py: %.ui
+	$(PYUIC) -x -o $@ $<
+
 INCLUDES = -Iip -Iio/hpmud -Iscan/sane 
 
 hplipdir = $(datadir)/hplip
diff -urNad hplip-2.8.9~/configure.in hplip-2.8.9/configure.in
--- hplip-2.8.9~/configure.in	2008-09-17 03:19:44.000000000 +1000
+++ hplip-2.8.9/configure.in	2008-10-17 13:09:19.000000000 +1100
@@ -336,6 +336,8 @@
    AC_MSG_RESULT("using $PYTHONINCLUDEDIR")
    AC_ARG_VAR(PYTHONINCLUDEDIR, [path to Python.h C header file])
    AC_CHECK_HEADERS(python$PYTHON_VERSION/Python.h, ,[AC_MSG_ERROR([cannot find python-devel support], 6)])
+   AC_ARG_VAR([PYUIC], [PyQT pyuic .ui to .py compiler command])
+   AC_CHECK_TOOLS([PYUIC], [pyuic])
 fi
 
 if test "$hpijs_only_build" = "no" && test "$scan_build" = "yes"; then
